src/helpers/parsing/child-fields/index.ts

```diff
@@ -0,0 +1,79 @@
+import { FieldNode, GraphQLInterfaceType, GraphQLObjectType, GraphQLSchema, InlineFragmentNode } from "graphql";
+import { castArray } from "lodash";
+import { unwrapInlineFragments } from "../inline-fragments";
+import { getKind } from "../kind";
+import { getName } from "../name";
+import { ParentNode } from "../types";
+
+export function addChildField(
+  node: ParentNode,
+  field: FieldNode,
+  schema: GraphQLSchema,
+  resourceID: string,
+): void {
+  if (!node.selectionSet) return;
+  const childFields = node.selectionSet.selections;
+  let added = false;
+
+  for (const childField of childFields) {
+    if (getKind(childField) === "InlineFragment") {
+      const inlineFragmentNode = childField as InlineFragmentNode;
+
+      if (inlineFragmentNode.typeCondition) {
+        const name = getName(inlineFragmentNode.typeCondition);
+
+        if (name) {
+          const type = schema.getType(name);
+
+          if (type instanceof GraphQLObjectType || type instanceof GraphQLInterfaceType) {
+            const objectOrInterfaceType = type as GraphQLObjectType | GraphQLInterfaceType;

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)